test/extension-tests/successful-build/test/prefered-generators.test.ts

```diff
@@ -0,0 +1,213 @@
+import {ITestCallbackContext, IHookCallbackContext} from 'mocha';
+
+interface BuildSystemConfiguration {
+  defaultKit: string;
+  expectedDefaultGenerator: string;
+  path?: string;
+}
+
+let workername = process.env.APPVEYOR_BUILD_WORKER_IMAGE;
+
+if (process.env.TRAVIS_OS_NAME) {
+  workername = process.env.TRAVIS_OS_NAME;
+}
+
+if (workername === undefined) {
+  workername = 'DevPC';
+}
+
+const DefaultCompilerMakeSystem: {[os: string]: BuildSystemConfiguration[]} = {
+  ['DevPC']: [
+    {defaultKit: 'VisualStudio.14.0', expectedDefaultGenerator: 'Visual Studio 14 2015'},
+    {
+      defaultKit: 'GCC',
+      expectedDefaultGenerator: 'MinGW Makefiles',
+      path: 'C:\\Program Files\\mingw-w64\\x86_64-7.2.0-posix-seh-rt_v5-rev1\\mingw64\\bin'
+    },
+    {defaultKit: 'Clang', expectedDefaultGenerator: 'MinGW Makefiles', path: 'C:\\Program Files\\LLVM\\bin'}
+  ],
+  ['Visual Studio 2017']: [
+    {defaultKit: 'Visual Studio Community 2017', expectedDefaultGenerator: 'Visual Studio 15 2017'},
+    {
+      defaultKit: 'GCC',
+      expectedDefaultGenerator: 'MinGW Makefiles',
+      path: 'C:\\mingw-w64\\x86_64-7.2.0-posix-seh-rt_v5-rev1\\mingw64\\bin'
+    }
+  ],
+  ['Visual Studio 2017 Preview']: [
+    {defaultKit: 'Visual Studio Community 2017', expectedDefaultGenerator: 'Visual Studio 15 2017'}
+  ],
+  ['Visual Studio 2015']: [
+    {defaultKit: 'VisualStudio.14.0', expectedDefaultGenerator: 'Visual Studio 14 2015'},
+    {
+      defaultKit: 'GCC',
+      expectedDefaultGenerator: 'MinGW Makefiles',
+      path: 'C:\\mingw-w64\\x86_64-6.3.0-posix-seh-rt_v5-rev1\\mingw64\\bin'
+    }
+  ],
+  ['Visual Studio 2013']: [
+    {defaultKit: 'VisualStudio.11.0', expectedDefaultGenerator: 'Visual Studio 11 2012'},
+    {defaultKit: 'GCC', expectedDefaultGenerator: 'MinGW Makefiles', path: 'C:\\MinGW\\bin'}
+  ],
+  ['linux']: [
+    {defaultKit: 'Clang', expectedDefaultGenerator: 'Unix Makefiles'},
+    {defaultKit: 'GCC', expectedDefaultGenerator: 'Unix Makefiles'}
+  ],
+  ['osx']: [{
+    defaultKit: 'Apple',
+    expectedDefaultGenerator: 'Unix Makefiles',
+    path:
+        '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin;/Applications/Xcode.app/Contents/Developer/usr/bin'
+  }]
+};
+
+interface CmakeContext {
+  cmt: CMakeTools;
+  testEnv: DefaultEnvironment;
+  pathBackup: string;
+  buildsystem: BuildSystemConfiguration;
+}
+
+function isPreferedGeneratorInKit(defaultKit: string): boolean { return RegExp('^VisualStudio').test(defaultKit); }
+
+function skipTestWithoutPreferredGeneratorInKit(testContext: any, context: CmakeContext): void {
+  if (!isPreferedGeneratorInKit(context.buildsystem.defaultKit)) {
+    testContext.skip();
+  }
+}
+
+// function skipTestWithPreferredGeneratorInKit(testContext: any, context :CmakeContext): void {
+//   if (isPreferedGeneratorInKit(context.buildsystem.defaultKit)) {
+//     testContext.skip();
+//   }
+// }
```

Comment:
Dead code?

Reply:
No it is used by tests,  but they are comment out, because the expected behavior is not clear.  We need to clarify how error messages provided to the user.